tools/pre-commit/pre-commit-check-versions.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+"""
+Helper to check if the tool versions specified in the pre-commit config file
+match the general project requirements.
+"""
+
+import sys
+from dataclasses import dataclass, field
+from enum import Enum
+from pathlib import Path
+from typing import Dict, List
+
+import click
+import yaml
+from packaging.requirements import Requirement
+from packaging.version import Version, parse
+
+PRE_COMMIT_CONFG_NAME = ".pre-commit-config.yaml"
+REQUIREMENTS_DEV_PATH = "../../requirements/requirements-dev.txt"
+
+
+@dataclass(frozen=True)
+class ToolConfig:
+    name: str
+    url: str
+
+
+class Tool(Enum):
+    ISORT = ToolConfig(name="isort", url="git://github.com/pre-commit/mirrors-isort")
+    BLACK = ToolConfig(name="black", url="https://github.com/python/black")
+    MYPY = ToolConfig(name="mypy", url="https://github.com/pre-commit/mirrors-mypy")
+    PYLINT = ToolConfig(name="pylint", url="https://github.com/pre-commit/mirrors-pylint")
+
+
+@dataclass
+class ToolVersion:
+    tool: Tool
+    version: Version
+    additional_deps: List[Requirement] = field(default_factory=list)
+
+
+_TOOL_BY_REPO_URL: Dict[str, Tool] = {tool.value.url: tool for tool in Tool}
+
+
+def find_pre_commit_config_file() -> Path:
+    parents = Path(__file__).resolve().parent.parents
+    for parent in parents:
+        config_file = parent.joinpath(PRE_COMMIT_CONFG_NAME)
+        if config_file.exists():
+            return config_file
+    raise ValueError(f"No pre-commit config file found any of {parents}.")
```

Comment:
This could probably be more stable (against being moved around) if it were to search from git-root down. But should be fine as is!

Reply:
Good point. Done.